CMakeLists.txt

```diff
@@ -16,6 +16,7 @@ SET_PROPERTY(GLOBAL PROPERTY ALLOW_DUPLICATE_CUSTOM_TARGETS 1)
 #message(STATUS "CMAKE_${LANGUAGE}_COMPILER_WORKS = ${CMAKE_${LANGUAGE}_COMPILER_WORKS}")
 
 include_directories(${PROJECT_SOURCE_DIR})
+include_directories(${PROJECT_BINARY_DIR})
```

Comment:
This line is not good here. Unfortunately there are 2 config.h files, one in build root from adios, and one in src/mxml from the mxml configuration. The idea was not to modify the included mxml-2.9 source at all. 
I get a lot of warnings in compiling mxml because of this line. 
I also added this line to tests/test_src/CMakeLists.txt today because it is needed there for a recently updated test. Let me know if any other directories need this line as well to build on Apple.

Reply:
I meant, I committed a change in master today for tests/test_src/CMakeLists.txt

Reply:
To summarize, I removed this line in my branch, and used the latest commit in master, and now I can build on linux without any warning.

Reply:
When I remove the `PROJECT_BINARY_DIR` (and build with CMake) I get the following error
```
/usr/local/src/adios/src/core/adios_transport_hooks.h:11:10: fatal error: 'config.h' file not found
#include "config.h"
         ^~~~~~~~~~
```

@pnorbert are you exclusively doing in-source Makefile builds?